tcl/pd_guiprefs.tcl

```diff
@@ -112,14 +112,17 @@ proc ::pd_guiprefs::init {} {
                 if {[catch {exec defaults write $adomain $akey -array} errorMsg]} {
                     ::pdwindow::error "write_config $akey: $errorMsg\n"
                 }
+                # invoke /bin/sh -c to pass command as a string, this ensures
+                # the shell interprets the single quotes around the value so the
+                # quotes don't end up being saved as part of the string itself
                 if {$arr} {
                     foreach filepath $data {
                         set escaped [escape_for_plist $filepath]
-                        exec defaults write $adomain $akey -array-add $escaped
+                        exec /bin/sh -c "defaults write $adomain $akey $escaped"
```

Comment:
I went ahead and added it. I tried a number of different quoting schemes but invoking the shell was the only one that worked. Definitely not ideal but was a suggestion I found on StackOverflow.